data-prepper-api/src/main/java/org/opensearch/dataprepper/expression/ExpressionEvaluator.java

```diff
@@ -30,16 +30,13 @@ default Boolean evaluateConditional(final String statement, final Event context)
         Object result;
         try {
             result = evaluate(statement, context);
-            if (result == null) {
-                return false;
-            }
             if (result instanceof Boolean) {
                 return (Boolean) result;
             }
+            throw new ClassCastException("Unexpected expression return value of " + result);
```

Comment:
Since this PR raises the idea of other exceptions, this is a good candidate for a better exception.

Declare:

```
public class UnexpectedEvaluatedReturnTypeException extends ExpressionEvaluationException {
```

And then throw:

```
throw new UnexpectedEvaluatedReturnTypeException("Unexpected expression return value of " + result);
```

We don't have to do for this PR, but it would be a good improvement.